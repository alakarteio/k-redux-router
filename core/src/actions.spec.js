/* eslint-env jest */
import {
  push,
  replace,
  goBack,
  goForward,
} from './actions'

describe('actions', () => {
  const match = result => () => expect(result).toMatchSnapshot()

  describe('push', () => {
    it('should push the user route with no param', match(push('user')))
    it('should push the user route with path params', match(push('user', { id: 2, name: 'Delphine' })))
    it('should push the user route with path and query params', match(push('user', { id: 3 }, { query: 'param', token: 'dzejfzeoif' })))
  })

  describe('replace', () => {
    it('should replace with user route with no param', match(replace('user')))
    it('should replace with user route with path params', match(replace('user', { id: 2, name: 'Delphine' })))
    it('should replace with user route with path and query params', match(replace('user', { id: 3 }, { query: 'param', token: 'dezijfeziofz' })))
  })

  describe('goBack', () => {
    it('should go to one step back', match(goBack()))
    it('shuld go to 3 steps back', match(goBack(3)))
  })

  describe('goForward', () => {
    it('should go to one step forward', match(goForward()))
    it('should go to 3 steps forward', match(goForward(3)))
  })
})
