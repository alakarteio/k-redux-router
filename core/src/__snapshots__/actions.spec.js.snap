// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions goBack should go to one step back 1`] = `
Object {
  "payload": 1,
  "type": "@@router/GO_BACK",
}
`;

exports[`actions goBack shuld go to 3 steps back 1`] = `
Object {
  "payload": 3,
  "type": "@@router/GO_BACK",
}
`;

exports[`actions goForward should go to 3 steps forward 1`] = `
Object {
  "payload": 3,
  "type": "@@router/GO_FORWARD",
}
`;

exports[`actions goForward should go to one step forward 1`] = `
Object {
  "payload": 1,
  "type": "@@router/GO_FORWARD",
}
`;

exports[`actions push should push the user route with no param 1`] = `
Object {
  "payload": Object {
    "code": "user",
    "params": Object {
      "path": undefined,
      "query": undefined,
    },
  },
  "type": "@@router/PUSH",
}
`;

exports[`actions push should push the user route with path and query params 1`] = `
Object {
  "payload": Object {
    "code": "user",
    "params": Object {
      "path": Object {
        "id": 3,
      },
      "query": Object {
        "query": "param",
        "token": "dzejfzeoif",
      },
    },
  },
  "type": "@@router/PUSH",
}
`;

exports[`actions push should push the user route with path params 1`] = `
Object {
  "payload": Object {
    "code": "user",
    "params": Object {
      "path": Object {
        "id": 2,
        "name": "Delphine",
      },
      "query": undefined,
    },
  },
  "type": "@@router/PUSH",
}
`;

exports[`actions replace should replace with user route with no param 1`] = `
Object {
  "payload": Object {
    "code": "user",
    "params": Object {
      "path": undefined,
      "query": undefined,
    },
  },
  "type": "@@router/REPLACE",
}
`;

exports[`actions replace should replace with user route with path and query params 1`] = `
Object {
  "payload": Object {
    "code": "user",
    "params": Object {
      "path": Object {
        "id": 3,
      },
      "query": Object {
        "query": "param",
        "token": "dezijfeziofz",
      },
    },
  },
  "type": "@@router/REPLACE",
}
`;

exports[`actions replace should replace with user route with path params 1`] = `
Object {
  "payload": Object {
    "code": "user",
    "params": Object {
      "path": Object {
        "id": 2,
        "name": "Delphine",
      },
      "query": undefined,
    },
  },
  "type": "@@router/REPLACE",
}
`;
